File: src/pages/exercises/function-figure-text/_codes/php/response/spec/figure-text.spec.php
<?php

require __DIR__ . '/../src/figure-text.php';

describe('Figure Text Tool', function () {

  it('making triangle text with size 1', function () {
    expect(triangleText(1))->toBe(
      '#'
    );
  });

  it('making triangle text with size 2', function () {
    expect(triangleText(2))->toBe(
      "# \n" .
        '##'
    );
  });

  it('making triangle text with size 3', function () {
    expect(triangleText(3))->toBe(
      "#  \n" .
        "## \n" .
        '###'
    );
  });

  it('making half diamond text with size 1', function () {
    expect(halfDiamondText(1))->toBe(
      '#'
    );
  });

  it('making half diamond text with size 2', function () {
    expect(halfDiamondText(2))->toBe(
      "# \n" .
        "##\n" .
        '# '
    );
  });

  it('making half diamond text with size 3', function () {
    expect(halfDiamondText(3))->toBe(
      "#  \n" .
        "## \n" .
        "###\n" .
        "## \n" .
        '#  '
    );
  });

  it('making diamond text with size 1', function () {
    expect(diamondText(1))->toBe(
      '#'
    );
  });

  it('making diamond text with size 2', function () {
    expect(diamondText(2))->toBe(
      " # \n" .
        "###\n" .
        ' # '
    );
  });

  it('making diamond text with size 3', function () {
    expect(diamondText(3))->toBe(
      "  #  \n" .
        " ### \n" .
        "#####\n" .
        " ### \n" .
        '  #  '
    );
  });

  it('making board text with size 1', function () {
    expect(boardText(1))->toBe(
      '#'
    );
  });

  it('making board text with size 2', function () {
    expect(boardText(2))->toBe(
      "# \n" .
        ' #'
    );
  });

  it('making board text with size 3', function () {
    expect(boardText(3))->toBe(
      "# #\n" .
        " # \n" .
        '# #'
    );
  });

  it('making board text with size 4', function () {
    expect(boardText(4))->toBe(
      "# # \n" .
        " # #\n" .
        "# # \n" .
        ' # #'
    );
  });
});

File: src/pages/exercises/function-figure-text/_codes/ruby/code/figure-text.spec.rb
require_relative "figure-text.rb"

describe "Figure Text Tool" do
  context "making triangle text with size 1" do
    it "#" do
      expect(triangle_text(1)).to eql("#")
    end
  end

  context "making triangle text with size 2" do
    it "# \n" + "##" do
      expect(triangle_text(2)).to eql("# \n" + "##")
    end
  end

  context "making triangle text with size 3" do
    it "#  \n" + "## \n" + "###" do
      expect(triangle_text(3)).to eql("#  \n" + "## \n" + "###")
    end
  end

  context "making half diamond text with size 1" do
    it "#" do
      expect(half_diamond_text(1)).to eql("#")
    end
  end

  context "making half diamond text with size 2" do
    it "# \n" + "##\n" + "# " do
      expect(half_diamond_text(2)).to eql("# \n" + "##\n" + "# ")
    end
  end

  context "making half diamond text with size 3" do
    it "#  \n" + "## \n" + "###\n" + "## \n" + "#  " do
      expect(half_diamond_text(3)).to eql("#  \n" + "## \n" + "###\n" + "## \n" + "#  ")
    end
  end

  context "making diamond text with size 1" do
    it "#" do
      expect(diamond_text(1)).to eql("#")
    end
  end

  context "making diamond text with size 2" do
    it " # \n" + "###\n" + " # " do
      expect(diamond_text(2)).to eql(" # \n" + "###\n" + " # ")
    end
  end

  context "making diamond text with size 3" do
    it "  #  \n" + " ### \n" + "#####\n" + " ### \n" + "  #  " do
      expect(diamond_text(3)).to eql("  #  \n" + " ### \n" + "#####\n" + " ### \n" + "  #  ")
    end
  end

  context "making board text with size 1" do
    it "#" do
      expect(board_text(1)).to eql("#")
    end
  end

  context "making board text with size 2" do
    it "# \n" + " #" do
      expect(board_text(2)).to eql("# \n" + " #")
    end
  end

  context "making board text with size 3" do
    it "# #\n" + " # \n" + "# #" do
      expect(board_text(3)).to eql("# #\n" + " # \n" + "# #")
    end
  end

  context "making board text with size 4" do
    it "# # \n" + " # #\n" + "# # \n" + " # #" do
      expect(board_text(4)).to eql("# # \n" + " # #\n" + "# # \n" + " # #")
    end
  end
end

